src/linker/Linker/CompilerGeneratedState.cs

```diff
@@ -92,6 +104,7 @@ void ProcessMethod (MethodDefinition method)
 						Debug.Assert (stateMachineType.DeclaringType == type ||
 							(CompilerGeneratedNames.IsGeneratedMemberName (stateMachineType.DeclaringType.Name) &&
 							 stateMachineType.DeclaringType.DeclaringType == type));
+						callGraph.TrackCall (method, stateMachineType);
 						if (!_compilerGeneratedTypeToUserCodeMethod.TryAdd (stateMachineType, method)) {
 							var alreadyAssociatedMethod = _compilerGeneratedTypeToUserCodeMethod[stateMachineType];
 							_context.LogWarning (new MessageOrigin (method), DiagnosticId.MethodsAreAssociatedWithStateMachine, method.GetDisplayName (), alreadyAssociatedMethod.GetDisplayName (), stateMachineType.GetDisplayName ());
```

Comment:
I see that we already have a problematic warning like the one you're adding. Maybe just file an issue that these warnings are not actionable - especially if we don't have a way to actually hit it.

Reply:
https://github.com/dotnet/linker/issues/2726